keps/sig-apps/sidecarcontainers.md

```diff
@@ -284,3 +290,114 @@ One alternative would be to have a new field in the Pod Spec of `sidecarContaine
 Another alternative would be to change the Job Spec to have a `primaryContainer` field to tell it which containers are important. However I feel this is perhaps too specific to job when this Sidecar concept could be useful in other scenarios.
 
 A boolean flag of `sidecar: true` could be used to indicate which pods are sidecars, however this prevents additional ContainerLifecycles from being added in the future.
+
+## Production Readiness Questionnaire
+
+#### Feature enablement and rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate
+	  - Feature gate name: `SidecarLifecycle`
+    - Components depending on the feature gate: `Kubelet`, `ApiServer`
+  - [ ] Other
+    - Describe the mechanism:
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**  
+  The feature can be disabled, doing so would cause any pods that use the sidecar lifecycle to revert back to using the standard pod lifecycle that everything currently obeys.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+
+  Any pods that were using the sidecar lifecycle would now start following that lifecycle again.
+
+* **Are there any tests for feature enablement/ disablement?**  
+  Not currently
+
+#### Scalability
+
+* **Will enabling / using this feature result in any new API calls?**  
+  No
+
+* **Will enabling / using this feature result in introducing new API types?**  
+  No
+* **Will enabling / using this feature result in any new calls to cloud
+  provider?**
+
+  No
+* **Will enabling / using this feature result in increasing size or count
+  of the existing API objects?**  
+  Describe them providing:
+  - API type(s): Pod
+  - Estimated increase in size: 13B
+  - Estimated amount of new objects: Field is optional
+
+
+* **Will enabling / using this feature result in increasing time taken by any
+  operations covered by [existing SLIs/SLOs][]?**  
+  Pods using the sidecar lifecycle will take longer for all containers to start as they will wait for the sidecars to start before starting non sidecar containers, this is by design.
```

Comment:
Even though this is by-design, this means that we may not be able to meet the SLO, which is really bad.
That said, given we want this feature, the action from this would be to update the SLO.
I think that we should basically specify that the SLO is valid only for pods with not sidecar containers for now, and leave a TODO to rethink that later.
Can you please open an issue in community repo for that?

@kubernetes/sig-scalability-api-reviews

Reply:
I've opened an issue here https://github.com/kubernetes/community/issues/4454